googletest/src/gtest.cc

```diff
@@ -1629,10 +1629,7 @@ AssertionResult CmpHelper##op_name(const char* expr1, const char* expr2, \
   if (val1 op val2) {\
     return AssertionSuccess();\
   } else {\
-    return AssertionFailure() \
-        << "Expected: (" << expr1 << ") " #op " (" << expr2\
-        << "), actual: " << FormatForComparisonFailureMessage(val1, val2)\
-        << " vs " << FormatForComparisonFailureMessage(val2, val1);\
+    return CmpHelperOpFailure(expr1, expr2, val1, val2, #op);\
```

Comment:
What would you think of using the ternary operator, inside the macro definition? As in:

    return (val1 op val2) ? AssertionSuccess() : CmpHelperOpFailure(expr1, expr2, val1, val2, #op); \

Reply:
I think I would stick to if-else since it's easier to debug and matches [style of the same macro in the header file]( https://github.com/kuzkry/googletest/blob/c5debf4680bb933e14794e0a21547e6dc5c1364d/googletest/include/gtest/gtest.h#L1627).

Reply:
Ah, I see now. The two `GTEST_IMPL_CMP_HELPER_` macro definitions used to be identical copies, but they got "out of sync" by commit https://github.com/google/googletest/commit/6884259b7d57bc8d70771f19c7d492e89d2bfa9a ("Reduce the stack frame size for CmpHelper* functions...", November 17, 2014), which adjusted the macro in gtest.h, but not the one in gtest.cc.